### ISSUE TITLE
Download of Sentinel-3 OLCI failing

### ISSUE THREAD

**Issue by fsteinmetz** [sentinelsat/sentinelsat#286]
I am trying to download Sentinel-3 OLCI files on coda.eumetsat.int using SentinelSat but I get failures with some files (it works for `'9b8485ec-e7dc-4fb9-8b42-5a966990f0ca'`, for instance) :

```python
api = SentinelAPI(user, password, 'https://coda.eumetsat.int')
api.download('0c7113b6-7047-420c-854a-9524e7a8e307')
```
```
Traceback (most recent call last):
  File "download.py", line 37, in <module>
    api.download('0c7113b6-7047-420c-854a-9524e7a8e307')
  File "/home/applis/anaconda/envs/py3v19.04/lib/python3.6/site-packages/sentinelsat/sentinel.py", line 493, in download
    product_info = self.get_product_odata(id)
  File "/home/applis/anaconda/envs/py3v19.04/lib/python3.6/site-packages/sentinelsat/sentinel.py", line 428, in get_product_odata
    values = _parse_odata_response(response.json()['d'])
  File "/home/applis/anaconda/envs/py3v19.04/lib/python3.6/site-packages/sentinelsat/sentinel.py", line 1126, in _parse_odata_response
    'footprint': _parse_gml_footprint(product["ContentGeometry"]),
  File "/home/applis/anaconda/envs/py3v19.04/lib/python3.6/site-packages/sentinelsat/sentinel.py", line 1049, in _parse_gml_footprint
    geometry_xml = ET.fromstring(geometry_str)
  File "/home/applis/anaconda/envs/py3v19.04/lib/python3.6/xml/etree/ElementTree.py", line 1314, in XML
    parser.feed(text)
TypeError: a bytes-like object is required, not 'NoneType'
```

**Comment by valgur** [comment]
This looks like mostly a server-side issue, but we can try to handle it better inside sentinelsat as well.

The [minimal product OData info](https://coda.eumetsat.int/odata/v1/Products('0c7113b6-7047-420c-854a-9524e7a8e307')?$format=json) for this product, which is queried from the server before starting a download, returns `null` for the footprint geometry for some reason.
However, surprisingly, the full OData attributes for this product still has a [`Footprint` attribute](https://coda.eumetsat.int/odata/v1/Products('0c7113b6-7047-420c-854a-9524e7a8e307')/Attributes('Footprint')) withthe correct footprint geometry.

I suggest reporting this issue to coda.eumetsat.int and/or the [Data Hub System](https://sentineldatahub.github.io/DataHubSystem) developers.

Meanwhile, on the sentinelsat side a possible workaround would be to use the 'Footprint' attribute when doing a `full=True` OData query and the normal footprint field is empty. For a `full=False` we could do an extra query to `Product('<uuid>')/Attributes('Footprint')` to get that info.

**Comment by fsteinmetz** [comment]
Thanks, I have reported to coda.eumetsat.int

**Comment by Fernerkundung** [comment]
> The minimal product OData info for this product, which is queried from the server before starting a download, returns null for the footprint geometry for some reason.

@valgur 
Would it make sense to make sentinelsat parse this as `str('null')` instead of returning `None` and also throw an exception?

The workaround you describe may work in this particular case, but we can't be sure for all the other endpoints. I'd favour a exception with a clear error message over an "automagic" solution that has the potential to silently fail.

**Comment by valgur** [comment]
I mostly agree, but I would return `None` as a footprint rather than raise an exception to not needlessly break product downloads as is happening here. The typical CLI usage with `--footprints` uses the OpenSearch result anyway (which in this case has the correct footprint value).

> Would it make sense to make sentinelsat parse this as str('null') instead of returning None ...?
The `null` JSON value gets converted into `None` automatically anyway and seems like a more meaningful and convenient to parse return value to me.

**Comment by Fernerkundung** [comment]
> I mostly agree, but I would return None as a footprint rather than raise an exception to not needlessly break product downloads as is happening here. 

Completely forgot that this would break all other downloads in that search. Fully agree with your proposition :+1:

**Comment by j08lue** [comment]
Closed with #325
